pkg/gds/admin_test.go

```diff
@@ -86,6 +88,41 @@ func (s *gdsTestSuite) doRequest(handle gin.HandlerFunc, c *gin.Context, w *http
 	return res
 }
 
+// Test that the middleware returns the corect error when making unauthenticated
+// requests to protected endpoints.
+func (s *gdsTestSuite) TestAuthenticationMiddleware() {
+	s.svc.GetAdmin().SetHealth(true)
+	for _, endpoint := range []struct {
+		name   string
+		method string
+		path   string
+		status int
+	}{
+		{"authenticate", http.MethodPost, "/v2/authenticate", http.StatusForbidden},
+		{"summary", http.MethodGet, "/v2/summary", http.StatusUnauthorized},
+		{"autocomplete", http.MethodGet, "/v2/autocomplete", http.StatusUnauthorized},
+		{"reviews", http.MethodGet, "/v2/reviews", http.StatusUnauthorized},
+		{"listVASPs", http.MethodGet, "/v2/vasps", http.StatusUnauthorized},
+		{"retrieveVASP", http.MethodGet, "/v2/vasps/42", http.StatusUnauthorized},
+		{"review", http.MethodPost, "/v2/vasps/42/review", http.StatusUnauthorized},
+		{"resend", http.MethodPost, "/v2/vasps/42/resend", http.StatusUnauthorized},
+		{"listReviewNotes", http.MethodGet, "/v2/vasps/42/notes", http.StatusUnauthorized},
+		{"createReviewNote", http.MethodPost, "/v2/vasps/42/notes", http.StatusUnauthorized},
+		{"updateReviewNote", http.MethodPut, "/v2/vasps/42/notes/1", http.StatusUnauthorized},
+		{"deleteReviewNote", http.MethodDelete, "/v2/vasps/42/notes/1", http.StatusUnauthorized},
+	} {
+		s.T().Run(endpoint.name, func(t *testing.T) {
+			serv := httptest.NewServer(s.svc.GetAdmin().Routes())
+			defer serv.Close()
```

Comment:
Any reason you're running the server inside of the test closure? Servers are designed to handle multiple requests in isolation, I think it might be better to extract this to the outer function scope just to make the tests a bit more efficient.